### DISCUSSION TITLE
failing to get Python examples working (on Mac)

### DISCUSSION THREAD

**Discussion by BillSeitz** [deepgram/community#500] [General help] [answered]
I suspect I did some installation wrong, or in the wrong order.

Now getting `invalid syntax` error on `match self.parent` - I suspect there are 2 copies of some pieces of library, and the wrong one is being called. But I don't want to make things worse....

Full log of attempts is at <http://webseitz.fluxent.com/wiki/Deepgram>

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by allenjiang666** [comment]
I have the same problem. The python sdk does work. I think there is a bug in the sdk.
Traceback (most recent call last):
  File "/Users/Allen/Projects/AI Tutor/test/test.py", line 3, in <module>
    from deepgram import Deepgram
  File "/Users/Allen/opt/anaconda3/lib/python3.9/site-packages/deepgram/__init__.py", line 9, in <module>
    from .client import DeepgramClient
  File "/Users/Allen/opt/anaconda3/lib/python3.9/site-packages/deepgram/client.py", line 125
    match self.parent:
          ^
SyntaxError: invalid syntax

**Reply by abdulhadi240** [reply]
same for me . sdk is not working . well i used deepgramclient with some changes and its working fine for now

**Comment by davidvonthenen** [comment]
Hi @abdulhadi240 and @BillSeitz 

We just released a major version of the Python SDK `v3`, which means doing a `pip install deepgram-sdk` will grab the latest version of the SDK. As you know, moving between major versions (ie `v2` -> `v3`) will always cause breaking interface changes by [semver](https://semver.org/) definition.

We have updated the README to include semver best-practices when developing projects, applications, etc using the Deepgram SDK (or any consumable library). You can read more about it here:
https://github.com/deepgram/deepgram-python-sdk?tab=readme-ov-file#installation

**Reply by abdulhadi240** [reply]
Great . But still when my url is for instance a YouTube url directing to a video. The code is giving error. On the other hand using audio is working perfectly. Can u guide me how to input a video url?

**Reply by davidvonthenen** [reply, marked as answer]
Someone else just posted this question... you can't pass the YouTube URL to Deepgram because that points to an HTML page. Do a `curl` command and take a look.

**Comment by jpvajda** [comment]
Related Discussion: https://github.com/orgs/deepgram/discussions/499

**Comment by BillSeitz** [comment]
I'm still getting the same error (syntax error on match self.parent). http://webseitz.fluxent.com/wiki/Deepgram

I have nasty feeling I've ended up with multiple copies of sdk/example files, I think I'm going to nuke this folder and start over.

is cloning still necessary, or only if you're (repeatedly) hitting YouTube URLs?

**Comment by BillSeitz** [comment]
Still failing same way http://webseitz.fluxent.com/wiki/Deepgram#:~:text=export%20DEEPGRAM_API_KEY

Are others having more success?

**Reply by davidvonthenen** [reply]
Version 3 of the SDK supports Python 3.10 or higher. From the logs in discord, it looks like you are running 3.8. I would update your python to a version under going active Python support or if you need to use 3.9 or lower, you can continue to use v2 of the SDK.

**Reply by davidvonthenen** [reply]
3.10 has been around for a while (since 2020) and is in maintenance at this point.

3.8 will be EOL this year, 3.9 will be EOL next year.

https://devguide.python.org/versions/

**Reply by davidvonthenen** [reply]
Some of the newer features required 3.10 and I think it's a fair trade off.

Someone was attempting to use 3.9 and ran into your exact same error.
